seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/SeaTunnelNodeContext.java

```diff
@@ -45,15 +50,28 @@ public NodeExtension createNodeExtension(@NonNull Node node) {
 
     @Override
     public Joiner createJoiner(Node node) {
+
         JoinConfig join =
                 getActiveMemberNetworkConfig(seaTunnelConfig.getHazelcastConfig()).getJoin();
         join.verify();
 
-        if (join.getTcpIpConfig().isEnabled()) {
+        // update for seatunnel, lite member can not become master node
+        if (join.getMulticastConfig().isEnabled() && node.multicastService != null) {
+            log.info("Using LiteNodeDropOutMulticast Multicast discovery");
+            return new LiteNodeDropOutMulticastJoiner(node);
+        } else if (join.getTcpIpConfig().isEnabled()) {
             log.info("Using LiteNodeDropOutTcpIpJoiner TCP/IP discovery");
             return new LiteNodeDropOutTcpIpJoiner(node);
+        } else if (node.getProperties().getBoolean(DISCOVERY_SPI_ENABLED)
+                || isAnyAliasedConfigEnabled(join)
+                || join.isAutoDetectionEnabled()) {
+            log.info("Using LiteNodeDropOutDiscoveryJoiner Discovery SPI");
+            return new LiteNodeDropOutDiscoveryJoiner(node);
         }
+        return null;
```

Comment:
why not return `super.createJoiner(node);`?

Reply:
Has cover all cases. Same like original implement.

![image](https://github.com/user-attachments/assets/cf898f72-4546-43ad-a479-df215d70b95b)